larq_zoo/literature/meliusnet.py

```diff
@@ -0,0 +1,240 @@
+from typing import Optional, Sequence, Union
+
+import larq as lq
+import tensorflow as tf
+from zookeeper import Field, factory
+
+from core import utils
+from core.model_factory import ModelFactory
+
+
+@factory
+class MeliusNetFactory(ModelFactory):
+    weights: Optional[str] = Field(None)
+
+    # overall architecture configuration
+    num_blocks: Sequence[int]
+    transition_features: Sequence[int]
+    name: str
+
+    # Some default layer arguments.
+    use_conv_bias: bool = Field(False)
+    batch_norm_momentum: float = Field(0.9)
+    kernel_initializer: Optional[Union[str, tf.keras.initializers.Initializer]] = Field(
+        "glorot_normal"
+    )
+    input_quantizer = Field("ste_sign")
+    kernel_quantizer = Field("ste_sign")
+    kernel_constraint = Field("weight_clip")
+
+    def pool(self, x):
+        return tf.keras.layers.MaxPool2D(2, strides=2, padding="same")(x)
+
+    def norm(self, x):
+        return tf.keras.layers.BatchNormalization(momentum=self.batch_norm_momentum)(x)
+
+    def act(self, x):
+        return tf.keras.layers.Activation("relu")(x)
+
+    def quant_conv(self, x, filters, kernel, strides=1):
+        return lq.layers.QuantConv2D(
+            filters,
+            kernel,
+            strides=strides,
+            padding="same",
+            use_bias=self.use_conv_bias,
+            input_quantizer=self.input_quantizer,
+            kernel_quantizer=self.kernel_quantizer,
+            kernel_constraint=self.kernel_constraint,
+            kernel_initializer=self.kernel_initializer,
+        )(x)
+
+    def group_conv(self, x, filters, kernel, groups):
+        assert filters % groups == 0
+        assert x.shape.as_list()[-1] % groups == 0
+
+        x_split = tf.split(x, groups, axis=-1)
+
+        y_split = []
+        for i in range(groups):
+            y_split.append(
+                tf.keras.layers.Conv2D(
+                    filters // groups,
+                    kernel,
+                    padding="same",
+                    use_bias=self.use_conv_bias,
+                    kernel_initializer=self.kernel_initializer,
+                )(x_split[i])
+            )
```

Comment:
Not sure if TensorFlow supports that, but I generally prefer something like this:
```suggestion
y_split = [
    tf.keras.layers.Conv2D(
        filters // groups,
        kernel,
        padding="same",
        use_bias=self.use_conv_bias,
        kernel_initializer=self.kernel_initializer,
    )(split)
    for split in x_split
]
```